cmd/erasure-healfile.go

```diff
@@ -117,13 +116,18 @@ func (s ErasureStorage) HealFile(staleDisks []StorageAPI, volume, path string, b
 			}
 		}
 
-		// advance the chunk offset to prepare for next loop
-		// iteration
-		chunkOffset += chunksize
-
-		// reconstruct data - this computes all data and parity shards
-		if err = s.ErasureDecodeDataAndParityBlocks(blocks); err != nil {
-			return f, err
+		if chunksize > 0 {
+			// Reedsolomon does not know how to encode/decode empty data,
+			// therefore, we should only proceed to reconstructing blocks
+			// if we are sure we have some real data waiting for us.
+			if err = s.ErasureDecodeDataAndParityBlocks(blocks); err != nil {
+				return f, err
+			}
+		} else {
```

Comment:
I think this else is not required. If `chunksize == 0` than `len(blocks[i]) == 0.

Reply:
> I think this else is not required. If chunksize == 0 than `len(blocks[i]) == 0.

Yeah.. but sometimes blocks[i] is empty and sometimes it is nil because the previous loop tries to exit as fast as possible. My policy is to make things consistent in a such random behavior to avoid bugs..

Reply:
I do not agree that it would be sometimes nil and sometimes empty. It should be always empty?! We never assign `nil` to a block, right?! However I think it's still fine for explicitness.